src/ansys/dpf/core/animator.py

```diff
@@ -152,6 +152,130 @@ def animation():
         self._plotter.close()
         return result
 
+    def animate_workflow_modal(
+        self,
+        loop_over,
+        mode_number,
+        workflow,
+        output_name,
+        input_name="loop_over",
+        save_as="",
+        frame_number=21,
+        **kwargs,
+    ):
+
+        indices = list(np.abs(np.linspace(-1, 1, frame_number)))
+        if save_as:
+            if save_as.endswith(".gif"):
+                self._plotter.open_gif(save_as)
+            else:  # pragma: no cover
+                kwargs_in = _sort_supported_kwargs(bound_method=self._plotter.open_movie, **kwargs)
+                try:
+                    self._plotter.open_movie(save_as, **kwargs_in)
+                except ImportError as e:
+                    if "imageio ffmpeg plugin you need" in e.msg:
+                        raise ImportError(
+                            "The imageio-ffmpeg library is required to save "
+                            "animations. Please install it first with the command "
+                            "'pip install imageio-ffmpeg'"
+                        )
+                    else:
+                        raise e
+        freq_kwargs = kwargs.pop("freq_kwargs", {})
+
+        # Camera position
+        cpos = kwargs.pop("cpos", None)
+        if cpos:
+            if isinstance(cpos[0][0], float):
+                cpos = [cpos] * len(indices)
+
+        workflow.connect(input_name, [mode_number])
+
+        initial_field = workflow.get_output(output_name, core.types.field)
+        max = float(np.max(initial_field.data))
+
+        def render_frame(frame):
+            self._plotter.clear()
+
+            field = workflow.get_output(output_name, core.types.field)
+
+            # TODO: deform_by is not accepted for niw
+            # deform = None
+            # if "deform_by" in workflow.output_names:
+            #     deform = workflow.get_output("deform_by", core.types.field)
+
+            ponderation = float(indices[frame])
+            op = core.operators.math.scale(
```

Comment:
@clatapie I don't understand why you add redo the render frame. If you had connect the scale Operator at the end of animate_mode and name it's input as "loop_over" would it not work? Maybe @PProfizi I didn't understand your design?

Reply:
I think this is what @PProfizi explained to me during our call. We decided to let it that way for now and to change it in another PR.